### PR TITLE
Check for null attribute

### PR DESCRIPTION
**Pull request by hibooboo2**
Works to address rancher/rancher#2191

### CODE REVIEW

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapIdentitySearchProvider.java [hunk 1]
```diff
@@ -378,7 +378,11 @@ private Identity attributesToIdentity(Attributes search){
                 externalId = (String) search.get(LdapConstants.DN).get();
             } else if (isType(search, LdapConstants.GROUP_OBJECT_CLASS.get())) {
                 externalIdType = LdapConstants.GROUP_SCOPE;
-                accountName = (String) search.get(LdapConstants.GROUP_NAME_FIELD.get()).get();
+                if (search.get(LdapConstants.GROUP_NAME_FIELD.get()) != null) {
+                    accountName = (String) search.get(LdapConstants.GROUP_NAME_FIELD.get()).get();
+                } else {
+                    accountName = "";
```
Comment (cjellick) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapIdentitySearchProvider.java]: Set to DN

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapIdentitySearchProvider.java [hunk 2]
```diff
@@ -378,7 +378,11 @@ private Identity attributesToIdentity(Attributes search){
                 externalId = (String) search.get(LdapConstants.DN).get();
```
Comment (cjellick) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapIdentitySearchProvider.java]: For completeness, do the same thing for user_name_field above.

**Comment by cjellick** [comment]
LGTM

**Comment by alena1108** [comment]
LGTM

**Comment by kaos** [comment]
I would like an intermediate attempt at using `name` as fallback if the requested name field is missing, before using `dn`, as that is rather verbose, and `name` is most likely usable.

**Pull request merged**